### ISSUE TITLE
Interpreter should have an option to disable foreign key checks

### ISSUE THREAD

**Issue by amir20001** [jOOQ/jOOQ#8105]
### Expected behavior and actual behavior:
should have the option to disable foreign key check while building out the schema, allowing for out of order table definition and circular references. 

### Steps to reproduce the problem : https://github.com/amir20001/jOOQ-mcve/tree/table-order-example
```
SET FOREIGN_KEY_CHECKS = 0;

CREATE TABLE `foo` (
  `id`     BIGINT(20) NOT NULL,
  `bar_id` BIGINT(20) NOT NULL,

  PRIMARY KEY (`id`),
  FOREIGN KEY (`bar_id`) REFERENCES `bar` (`id`)
);

CREATE TABLE `bar` (
  `id`   BIGINT(20)    NOT NULL,
  `name` VARCHAR(2500) NOT NULL,

  PRIMARY KEY (`id`)
);

SET FOREIGN_KEY_CHECKS = 1;

```

### Versions:

- jOOQ: 3.11.7
- Java: 1.8
- Database (include vendor): DDLDatabase

**Comment by lukaseder** [comment]
Thank you very much for your report. That's quite an interesting MySQL feature. Supporting it will be quite complex as it cannot be emulated easily on H2 (which is used behind the scenes to support incremental `DDLDatabase` script executions). We could extract all the constraint definitions from the `CREATE TABLE` statements and execute them separately when encountering `SET FOREIGN_KEY_CHECKS = 1`

The workaround for the time being is to do that manually, of course:

```sql
CREATE TABLE `foo` (
  `id`     BIGINT(20) NOT NULL,
  `bar_id` BIGINT(20) NOT NULL,

  PRIMARY KEY (`id`)
);

CREATE TABLE `bar` (
  `id`   BIGINT(20)    NOT NULL,
  `name` VARCHAR(2500) NOT NULL,

  PRIMARY KEY (`id`)
);

ALTER TABLE `foo` ADD CONSTRAINT `foo_bar_id` FOREIGN KEY (`bar_id`) REFERENCES `bar` (`id`)
```

**Comment by amir20001** [comment]
that makes sense thanks for the update.

**Comment by lukaseder** [comment]
In the meantime, we're almost ready to provide the infrastructure for this feature request. We can probably emulate the MySQL command in our new DDL interpreter, which will ship in jOOQ 3.13.

I'll see if this can be done, soon.

**Comment by lukaseder** [comment]
This feature should also be made available via a global `Settings`, which defines the default behaviour of the interpreter: #9508.

The command referenced in this issue here would then temporarily override a copy of the settings value.

**Comment by lukaseder** [comment]
In order to support this thoroughly, we'll first need a way to model `SET` commands in the query model, and in the parser. I'll implement this properly later: #9780.

**Comment by lukaseder** [comment]
This is now implemented for jOOQ 3.13. A few rules will apply (and might change):

- This feature is not implemented in the current H2 based simulated `DDLDatabase`. Only when we interpret the SQL ourselves, we can support this (unless H2 emulates this as well)
- The new `Settings.parseSetCommands` needs to be turned on. This is turned off by default, for now, as we might not know all the possible `SET` syntaxes in the parser.
